src/Couchbase.Lite.Shared/API/Sync/ReplicatorConfiguration.cs

```diff
@@ -211,49 +207,94 @@ public ReplicatorType ReplicatorType
         /// Gets or sets the replicator heartbeat keep-alive interval. 
-        /// The default Heartbeat is <c>5</c> min.
+        /// The default is null (5 min interval is applied). 
+        /// * <c>5</c> min interval is applied when Heartbeat is set to null.
+        /// * null will be returned when default <c>5</c> min interval is applied.
         /// </summary>
         /// <exception cref="ArgumentException"> 
         /// Throw if set the Heartbeat to less or equal to 0 full seconds.
         /// </exception>
-        public TimeSpan Heartbeat
+        public TimeSpan? Heartbeat
         {
-            get => Options.Heartbeat;
-            set => _freezer.PerformAction(() => Options.Heartbeat = value);
+            get 
+            {
+                if (Options.Heartbeat < 0)
+                    return null;
+                else
+                    return TimeSpan.FromSeconds(Options.Heartbeat); 
+            }
+
+            set
+            {
+                if (value != null) {
+                    long sec = value.Value.Ticks / TimeSpan.TicksPerSecond;
+                    if (sec > 0) {
+                        _freezer.PerformAction(() => Options.Heartbeat = sec);
+                    } else {
+                        throw new ArgumentException(CouchbaseLiteErrorMessage.InvalidHeartbeatInterval);
+                    }
+                }
+            }
         }
 
         /// <summary>
-        /// Max number of retry attempts. The retry attempts will reset
+        /// Gets or sets the Max number of retry attempts. The retry attempts will reset
         /// after the replicator is connected to a remote peer. 
-        /// The default MaxRetries is <c>9</c> for a single shot replicator and 
-        /// <see cref="Int32.MaxValue" /> for a continuous replicator.
-        /// Set the MaxRetries to 0 will result in no retry attempt.
+        /// The default is <c>0</c> (<c>10</c> for a single shot replicator or 
+        /// <see cref="int.MaxValue" /> for a continuous replicator is applied.)
+        /// * <c>10</c> for a single shot replicator or <see cref="int.MaxValue" /> for a 
+        /// continuous replicator is applied when user set MaxAttempts to 0.
+        /// * 0 will be returned when default <c>10</c> for a single shot replicator or 
+        /// <see cref="int.MaxValue" /> for a continuous replicator is applied.
+        /// * Setting the value to 1 means that the replicator will try connect once and 
+        /// the replicator will stop if there is a transient error.
         /// </summary>
         /// <exception cref="ArgumentException">
-        /// Throw if set the MaxRetries to a negative value.
+        /// Throw if set the MaxAttempts to a negative value.
         /// </exception>
-        public int MaxRetries
+        public int MaxAttempts
         {
-            get => Options.MaxRetries >= 0 ? Options.MaxRetries : _continuous ? int.MaxValue : 9;
+            get => Options.MaxRetries + 1;
```

Comment:
When we did not get key value in Options.MaxRetries, -1 is applied there. 
Since Max Attempts = Max Retries + 1,
- 0 will be returned and 10 for a single shot replicator or int.MaxValue for a continuous replicator is applied in LiteCore..
- When Options.MaxRetries is 0 and MaxAttempts is 1, we know replicator will try to connect once if replicator stop with and transient error.